exporters/failure/collector_azure_devops.py

```diff
@@ -0,0 +1,174 @@
+from pelorus.errors import FailureProviderAuthenticationError
+from pelorus.timeutil import parse_assuming_utc, second_precision
+from pelorus.utils import Url
+
+_DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+
+
+@define(kw_only=True)
+class AzureDevOpsFailureCollector(AbstractFailureCollector):
+    """
+    Azure DevOps implementation of a FailureCollector
+    """
+
+    token: str = field(
+        metadata=env_vars(*env_var_names.TOKEN) | log(REDACT),
+        repr=False,
+    )
+
+    tracker_api: Url = field(
+        metadata=env_vars("SERVER"),
+        converter=converters.optional(pass_through(Url, Url.parse)),
+    )
+
+    projects: set[str] = field(
+        factory=set, converter=comma_or_whitespace_separated(set)
+    )
+
+    work_item_type: set[str] = field(
+        factory=set,
+        converter=comma_or_whitespace_separated(set),
+        metadata=env_vars("AZURE_DEVOPS_TYPE"),
+    )
+
+    work_item_priority: set[str] = field(
+        factory=set,
+        converter=comma_or_whitespace_separated(set),
+        metadata=env_vars("AZURE_DEVOPS_PRIORITY"),
+    )
+
+    def __attrs_post_init__(self):
+        credentials = BasicAuthentication("", self.token)
+        connection = Connection(base_url=self.tracker_api.url, creds=credentials)
+        self.client = connection.clients_v6_0.get_work_item_tracking_client()
+
+    def get_work_items(self) -> List[WorkItem]:
+        logging.info("Getting work items")
+
+        try:
+            wiql = Wiql(query="Select [System.Id] From WorkItems")
+            wiql_results = self.client.query_by_wiql(wiql).work_items
+            return [
+                self.client.get_work_item(int(result.id)) for result in wiql_results
+            ]
+        except AzureDevOpsServiceError as error:
+            logging.error(FailureProviderAuthenticationError.auth_message)
+            raise FailureProviderAuthenticationError from error
+        except Exception as error:
+            # TODO test limitation 200?!?!
+            logging.error(error)  # pragma: no cover
+            raise  # pragma: no cover
+
+    def filter_by_type(self, work_item_type: str) -> bool:
+        if not self.work_item_type:
+            return True
+        return work_item_type in self.work_item_type
+
+    def filter_by_priority(self, priority: str) -> bool:
+        if not self.work_item_priority:
+            return True
+        return str(priority) in self.work_item_priority
+
+    def filter_by_project(self, project: str) -> bool:
+        if not self.projects:
+            return True
+        return project in self.projects
+
+    def get_app_name(self, work_item: WorkItem) -> str:
+        try:
+            labels: str = work_item.fields["System.Tags"]
+            labels = labels.split("; ")
+
+            label_text = self.app_label + "="
+
+            for label in labels:
+                if label_text in label:
+                    return label.replace(label_text, "")
```

Comment:
Could you add some comment for this function, what is it doing and why is it doing that way? I don't understand why some relpacement needs to be here.

Reply:
the label is `label=app_name`, we just want `app_name`, removing the unnecessary data